# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/nsDNSService2.h
# Commit: af0f14eed70e
# Full Hash: af0f14eed70edb2abb1ecd3cae4928387287d3e1
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1901990
# File Overlap Count: 2
# Description:
#   Bug 1901990 - Introduce a pref for setting a mock HTTPS RR, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213559
# ==============================================================================

diff -r 9f913cbd60ee -r af0f14eed70e netwerk/dns/nsDNSService2.h
--- a/netwerk/dns/nsDNSService2.h	Thu Jun 13 15:08:04 2024 +0200
+++ b/netwerk/dns/nsDNSService2.h	Fri Jun 14 09:11:49 2024 +0000
@@ -17,6 +17,7 @@
 #include "nsString.h"
 #include "nsTHashSet.h"
 #include "nsHashKeys.h"
+#include "mozilla/Atomics.h"
 #include "mozilla/Mutex.h"
 #include "mozilla/Attributes.h"
 #include "TRRService.h"
@@ -109,8 +110,8 @@
   RefPtr<nsHostResolver> mResolver;
   nsCOMPtr<nsIIDNService> mIDN;
 
-  // mLock protects access to mResolver, mLocalDomains, mIPv4OnlyDomains and
-  // mFailedSVCDomainNames
+  // mLock protects access to mResolver, mLocalDomains, mIPv4OnlyDomains,
+  // mFailedSVCDomainNames, and mMockHTTPSRRDomain.
   mozilla::Mutex mLock MOZ_UNANNOTATED{"nsDNSServer.mLock"};
 
   // mIPv4OnlyDomains is a comma-separated list of domains for which only
@@ -118,6 +119,8 @@
   // a per-domain basis and work around broken DNS servers. See bug 68796.
   nsCString mIPv4OnlyDomains;
   nsCString mForceResolve;
+  nsCString mMockHTTPSRRDomain;
+  mozilla::Atomic<bool, mozilla::Relaxed> mHasMockHTTPSRRDomainSet{false};
   bool mBlockDotOnion = false;
   bool mNotifyResolution = false;
   bool mOfflineLocalhost = false;